Incompatible with gcc 16

When compiling the repository with gcc 16.1.0, an error occurs.
<details>
<summary> Compile log: </summary>

```
FAILED: [code=1] src/filter/kaleid0sc0pe/CMakeFiles/kaleid0sc0pe.dir/kaleid0sc0pe.cpp.o 
D:\compile_ff\msys64\ucrt64\bin\ccache.exe  g++ -DHAS_INTEL_INTRINSICS -Dkaleid0sc0pe_EXPORTS -ID:/compile_ff/build/frei0r-git/include -D_FORTIFY_SOURCE=2 -fstack-protector-strong -mtune=generic -O2 -pipe -D__USE_MINGW_ANSI_STDIO=1 -O3 -DNDEBUG -std=gnu++11 -MD -MT src/filter/kaleid0sc0pe/CMakeFiles/kaleid0sc0pe.dir/kaleid0sc0pe.cpp.o -MF src\filter\kaleid0sc0pe\CMakeFiles\kaleid0sc0pe.dir\kaleid0sc0pe.cpp.o.d -o src/filter/kaleid0sc0pe/CMakeFiles/kaleid0sc0pe.dir/kaleid0sc0pe.cpp.o -c D:/compile_ff/build/frei0r-git/src/filter/kaleid0sc0pe/kaleid0sc0pe.cpp
In file included from D:/compile_ff/build/frei0r-git/src/filter/kaleid0sc0pe/kaleid0sc0pe.cpp:25:
D:/compile_ff/build/frei0r-git/src/filter/kaleid0sc0pe/kaleid0sc0pe.h:42:9: warning: '__SSE2__' redefined
   42 | #define __SSE2__
      |         ^~~~~~~~
<built-in>: note: this is the location of the previous definition
In file included from D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/cstring:47,
                 from D:/compile_ff/build/frei0r-git/src/filter/kaleid0sc0pe/kaleid0sc0pe.cpp:28:
D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/bits/version.h:2660:143: error: operator '&&' has no right operand
 2660 | # if (__cplusplus >  202302L) && _GLIBCXX_HOSTED && (__cpp_structured_bindings >= 202411L && __cpp_expansion_statements >= 202411L && __SSE2__)
      |                                                                                                                                               ^
In file included from D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/ratio:45,
                 from D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/bits/chrono.h:39,
                 from D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/mutex:45,
                 from D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/future:43,
                 from D:/compile_ff/build/frei0r-git/src/filter/kaleid0sc0pe/kaleid0sc0pe.cpp:30:
D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/bits/version.h:2660:143: error: operator '&&' has no right operand
 2660 | # if (__cplusplus >  202302L) && _GLIBCXX_HOSTED && (__cpp_structured_bindings >= 202411L && __cpp_expansion_statements >= 202411L && __SSE2__)
      |                                                                                                                                               ^
In file included from D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/bits/chrono.h:51:
D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/bits/version.h:2660:143: error: operator '&&' has no right operand
 2660 | # if (__cplusplus >  202302L) && _GLIBCXX_HOSTED && (__cpp_structured_bindings >= 202411L && __cpp_expansion_statements >= 202411L && __SSE2__)
      |                                                                                                                                               ^
In file included from D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/mutex:62:
D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/bits/version.h:2660:143: error: operator '&&' has no right operand
 2660 | # if (__cplusplus >  202302L) && _GLIBCXX_HOSTED && (__cpp_structured_bindings >= 202411L && __cpp_expansion_statements >= 202411L && __SSE2__)
      |                                                                                                                                               ^
In file included from D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/bits/basic_string.h:48,
                 from D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/string:58,
                 from D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/stdexcept:41,
                 from D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/system_error:45,
                 from D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/future:45:
D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/bits/version.h:2660:143: error: operator '&&' has no right operand
 2660 | # if (__cplusplus >  202302L) && _GLIBCXX_HOSTED && (__cpp_structured_bindings >= 202411L && __cpp_expansion_statements >= 202411L && __SSE2__)
      |                                                                                                                                               ^
In file included from D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/string:80:
D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/bits/version.h:2660:143: error: operator '&&' has no right operand
 2660 | # if (__cplusplus >  202302L) && _GLIBCXX_HOSTED && (__cpp_structured_bindings >= 202411L && __cpp_expansion_statements >= 202411L && __SSE2__)
      |                                                                                                                                               ^
In file included from D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/atomic:51,
                 from D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/bits/atomic_futex.h:37,
                 from D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/future:48:
D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/bits/version.h:2660:143: error: operator '&&' has no right operand
 2660 | # if (__cplusplus >  202302L) && _GLIBCXX_HOSTED && (__cpp_structured_bindings >= 202411L && __cpp_expansion_statements >= 202411L && __SSE2__)
      |                                                                                                                                               ^
[104/322] Building CXX object src/generator/dem0scene/CMakeFiles/plasma.dir/plasma.cpp.o
[105/322] Building CXX object src/generator/onecol0r/CMakeFiles/onecol0r.dir/onecol0r.cpp.o
[106/322] Building CXX object src/generator/partik0l/CMakeFiles/partik0l.dir/partik0l.cpp.o
ninja: build stopped: subcommand failed.
[1/217] Building CXX object src/filter/kaleid0sc0pe/CMakeFiles/kaleid0sc0pe.dir/kaleid0sc0pe.cpp.o
FAILED: [code=1] src/filter/kaleid0sc0pe/CMakeFiles/kaleid0sc0pe.dir/kaleid0sc0pe.cpp.o 
D:\compile_ff\msys64\ucrt64\bin\ccache.exe  g++ -DHAS_INTEL_INTRINSICS -Dkaleid0sc0pe_EXPORTS -ID:/compile_ff/build/frei0r-git/include -D_FORTIFY_SOURCE=2 -fstack-protector-strong -mtune=generic -O2 -pipe -D__USE_MINGW_ANSI_STDIO=1 -O3 -DNDEBUG -std=gnu++11 -MD -MT src/filter/kaleid0sc0pe/CMakeFiles/kaleid0sc0pe.dir/kaleid0sc0pe.cpp.o -MF src\filter\kaleid0sc0pe\CMakeFiles\kaleid0sc0pe.dir\kaleid0sc0pe.cpp.o.d -o src/filter/kaleid0sc0pe/CMakeFiles/kaleid0sc0pe.dir/kaleid0sc0pe.cpp.o -c D:/compile_ff/build/frei0r-git/src/filter/kaleid0sc0pe/kaleid0sc0pe.cpp
In file included from D:/compile_ff/build/frei0r-git/src/filter/kaleid0sc0pe/kaleid0sc0pe.cpp:25:
D:/compile_ff/build/frei0r-git/src/filter/kaleid0sc0pe/kaleid0sc0pe.h:42:9: warning: '__SSE2__' redefined
   42 | #define __SSE2__
      |         ^~~~~~~~
<built-in>: note: this is the location of the previous definition
In file included from D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/cstring:47,
                 from D:/compile_ff/build/frei0r-git/src/filter/kaleid0sc0pe/kaleid0sc0pe.cpp:28:
D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/bits/version.h:2660:143: error: operator '&&' has no right operand
 2660 | # if (__cplusplus >  202302L) && _GLIBCXX_HOSTED && (__cpp_structured_bindings >= 202411L && __cpp_expansion_statements >= 202411L && __SSE2__)
      |                                                                                                                                               ^
In file included from D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/ratio:45,
                 from D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/bits/chrono.h:39,
                 from D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/mutex:45,
                 from D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/future:43,
                 from D:/compile_ff/build/frei0r-git/src/filter/kaleid0sc0pe/kaleid0sc0pe.cpp:30:
D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/bits/version.h:2660:143: error: operator '&&' has no right operand
 2660 | # if (__cplusplus >  202302L) && _GLIBCXX_HOSTED && (__cpp_structured_bindings >= 202411L && __cpp_expansion_statements >= 202411L && __SSE2__)
      |                                                                                                                                               ^
In file included from D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/bits/chrono.h:51:
D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/bits/version.h:2660:143: error: operator '&&' has no right operand
 2660 | # if (__cplusplus >  202302L) && _GLIBCXX_HOSTED && (__cpp_structured_bindings >= 202411L && __cpp_expansion_statements >= 202411L && __SSE2__)
      |                                                                                                                                               ^
In file included from D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/mutex:62:
D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/bits/version.h:2660:143: error: operator '&&' has no right operand
 2660 | # if (__cplusplus >  202302L) && _GLIBCXX_HOSTED && (__cpp_structured_bindings >= 202411L && __cpp_expansion_statements >= 202411L && __SSE2__)
      |                                                                                                                                               ^
In file included from D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/bits/basic_string.h:48,
                 from D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/string:58,
                 from D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/stdexcept:41,
                 from D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/system_error:45,
                 from D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/future:45:
D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/bits/version.h:2660:143: error: operator '&&' has no right operand
 2660 | # if (__cplusplus >  202302L) && _GLIBCXX_HOSTED && (__cpp_structured_bindings >= 202411L && __cpp_expansion_statements >= 202411L && __SSE2__)
      |                                                                                                                                               ^
In file included from D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/string:80:
D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/bits/version.h:2660:143: error: operator '&&' has no right operand
 2660 | # if (__cplusplus >  202302L) && _GLIBCXX_HOSTED && (__cpp_structured_bindings >= 202411L && __cpp_expansion_statements >= 202411L && __SSE2__)
      |                                                                                                                                               ^
In file included from D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/atomic:51,
                 from D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/bits/atomic_futex.h:37,
                 from D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/future:48:
D:/compile_ff/msys64/ucrt64/include/c++/16.1.0/bits/version.h:2660:143: error: operator '&&' has no right operand
 2660 | # if (__cplusplus >  202302L) && _GLIBCXX_HOSTED && (__cpp_structured_bindings >= 202411L && __cpp_expansion_statements >= 202411L && __SSE2__)
      |                                                                                                                                               ^
ninja: build stopped: subcommand failed.
```
</details>

Seems to be a problem of `kaleid0sc0pe.h`. Also, it might be caused by this:https://github.com/gcc-mirror/gcc/commit/1cfdd868cea8583ba498fcc22690070d4cb24ee1 , which is introduced in gcc 16.1.0.

> [!NOTE]
> As I apply the patch #274 and the log shows, msys should have nothing to do with this issue.

--- a/src/filter/kaleid0sc0pe/kaleid0sc0pe.h
+++ b/src/filter/kaleid0sc0pe/kaleid0sc0pe.h
@@ -35,7 +35,7 @@
 #ifdef __SSE2__
 #undef __SSE2__
 #endif
-#else
+#elif !defined(__SSE2__)
 // set sse2 flag on windows
 #if _M_IX86_FP == 2 || _M_X64 == 100
 #ifndef _M_ARM

